compare_pams.Rmd

```diff
@@ -1,37 +1,294 @@
+
+``` {r company_ids, echo = FALSE}
+# Company IDs
+
+# compute unique Company IDs
+base_ids <- unique(base_pams$`Company ID`)
+compare_ids <- unique(comparison_pams$`Company ID`)
+
+# compute the number of IDs lost and gained
+base_count <- length(base_ids)
+lost_count <- length(setdiff(base_ids, compare_ids))
+lost_percent <- 100 *(1 - ((base_count-lost_count)/base_count))
+
+gained_count <- length(setdiff(compare_ids, base_ids))
+gained_percent <- 100 *(1 - ((base_count-gained_count)/base_count))
 
-# FIXME: There seem to already be differences in the PAMS datasets between 
-# 2021Q4 and 2022Q4 that prevent this function from working for both
+# state statistics
+if (gained_percent > company_id_threshold) {
+  warning(glue::glue("PAMS gained {round(gained_percent, 2)}% new Company IDs"))
+} else {
+  message(glue::glue("PAMS gained {round(gained_percent, 2)}% new Company IDs"))
+}
+
+if (lost_percent > company_id_threshold) {
+  warning(glue::glue("PAMS lost {round(lost_percent, 2)}% of original Company IDs"))
+} else {
+  message(glue::glue("PAMS lost {round(lost_percent, 2)}% of original Company IDs"))
+}
+
+```
+
+### Expectation 2: Sectoral Production Values
+
+We expect that production values, in aggregate, shouldn't differ by more than `r sectoral_production_threshold`%.
+
+``` {r sectoral_production, echo = FALSE}
+# Calculate the total production values for both datasets
+total_production_base <- base_pams %>% 
+  filter(`Is Ultimate Parent` == TRUE) %>% 
+  filter(consolidation_method == "Direct Ownership") %>% 
+  filter(value_type == "production") %>% 
+  group_by(`Asset Sector`) %>% 
+  summarize(value = sum(value))
+
+total_production_comparison <- comparison_pams %>% 
+  filter(`Is Ultimate Parent` == TRUE) %>% 
+  filter(consolidation_method == "Direct Ownership") %>% 
+  filter(value_type == "production") %>% 
+  group_by(`Asset Sector`) %>% 
+  summarize(value = sum(value))
+
+# Check if the expectation is met
+sector_expectation <- full_join(
+  total_production_base, 
+  total_production_comparison, 
+  by = "Asset Sector",
+  suffix = c("_base", "_comparison")
+) %>% 
+  mutate(
+    difference = 100 * abs(value_base - value_comparison) / value_base
+  )
+
+# Summary
+cat("\nExpectation 2: Production Values\n")
+for (sector in unique(sector_expectation$`Asset Sector`)) {
+  sector_difference <- filter(sector_expectation, `Asset Sector` == sector) %>% pull(difference)
+  
+  if (sector_difference > sectoral_production_threshold) {
+    warning(glue::glue("PAMS total {sector} Production changed by {round(sector_difference, 2)}%"))
+  } else {
+    message(glue::glue("PAMS total {sector} Production changed by {round(sector_difference, 2)}%"))
+  }
+}
+
+```
+
+### Expectation 3: Sectors
+
+We expect that the total possible options for Sector and Technology are identical between the two datasets. If there are any gained or dropped levels here, we want to know about it. 
+
+``` {r sectors}
```

Comment:
As CJ already pointed out, this section should likely be `echo = FALSE` and have alternatives displayed where the if condition is not met, like all other sections?

Reply:
seeing below that no else condition is needed... but echo remains a point